### DISCUSSION TITLE
select to remove duplicates

### DISCUSSION THREAD

**Discussion by carlosstenzel** [supabase/supabase#529] [Questions] [answered]
i need help removing duplicate data from a table, using select

**Comment by kiwicopple** [comment]
Hi @carlosstenzel. Thanks for dropping in. Can you share some details of you table structure? Also do you want to select the data without duplicates, or do you want to delete the duplicates and the select the data?

From the details you shared, I think you want to do a `select distinct`. Your best option will be to create a database view that doesn't have any duplicates, then you can query that.

As a general rule it's good to put in more effort to a question than you expect the responder will take. It's really beneficial to the community when questions are clear and have a lot of details - it makes searching better for future visitors 👍

**Comment by carlosstenzel** [comment]
i'm just trying to select without duplicates
```
supabase.from('pedidos').select('*').eq('user_id', id)....
```

the column with duplicate is called `payment_intent`, but I think it is better to make a filter to check if it already exists, before insert

**Comment by ftonato** [comment]
Hey @carlosstenzel, how are you?

Could you share more details of you table structure? It would be easier for us to provide an answer.
Below is an example of what I imagine you have or you're trying to do:

| Country_id | Country_title | State             |
|------------|---------------|-------------------|
| 1          | Canada        | Alberta           |
| 2          | Canada        | British Columbia |
| 3          | Canada        | Manitoba          |
| 4          | China         | Xinjiang          |


```sql
CREATE TABLE Countries (
    Country_id int,
    Country_title varchar(255),
    State varchar(255)
);

INSERT INTO Countries (Country_id, Country_title, State)
VALUES (1, "Canada", "Alberta"),
       (2, "Canada", "British Columbia"),
       (3, "Canada", "Manitoba"),
       (4, "China", "Xinjiang");

SELECT MIN(Country_id) AS Country_id, Country_title, State
FROM Countries
GROUP BY Country_title
```

The code above, returns exactly that:

```markdown
1|Canada|Alberta
4|China|Xinjiang
```

In fact, it is removing duplicate items from the column `Country_title`, returning only the first one found.

**Note:** If it is better for you, you can write in Portuguese and ping me that I will try to help you.

**Comment by carlosstenzel** [comment]
```
SELECT MIN(payment_intent) AS payment_intent
FROM pedidos
GROUP BY payment_intent
```

and

```
Select payment_intent
From pedidos 
Group by payment_intent
Having Count(*) > 1
```

return only the payment_intetn field, **adding another field results in an error

my table

![Screenshot_2021-01-17 Supabase](https://user-images.githubusercontent.com/3890516/104856072-19e81400-58ef-11eb-89cc-4d1eb529545c.png)


** Gostaria de selecionar todos os dados da tabela `pedidos`, excluindo as linhas onde o `payment_intent` ser repetido. Utilizando a biblioteca JS do supabase: ex: ``supabase.from('pedidos').select('*')...``

**Comment by ftonato** [comment, marked as answer]
Hey @carlosstenzel,

**[EN]**

I understand your problem and unfortunately we still don't support exactly what you want, but as @kiwicopple mentioned, you can create a `view` to remove duplicate items. I'll share the code with you:

**[PT]**

Acho que entendi seu problema e infelizmente ainda não suportamos exatamente o que você quer, mas como o @kiwicopple citou, você pode criar uma `view` para remover os itens duplicados. Vou compartilhar o código contigo:

```sql
CREATE or REPLACE VIEW pedidos_view AS
  SELECT DISTINCT ON (payment_intent)
    data_init,
    id,
    id_stripe,
    nome,
    payment_intent,
    payment_status,
    slug,
    user_id
FROM pedidos
ORDER BY payment_intent, id;
```
**[EN]**
And in your application, just do this:

**[PT]**
E na sua aplicação, basta fazer isso: 

```js
const fetchPedidos = async () => {
  const { data, error } = await supabase
    .from('pedidos_view')
    .select(`*`)
  
    return { data, error };
}
```

Please, let me know if it worked =)

**Reply by carlosstenzel** [reply]
Perfeito, funcionou exatamente como gostaria. Muito obrigado pela ajuda @ftonato


 👍 🥇
